### PR TITLE
bug: [PagoPa-2421] - advisory lock

### PR DESCRIPTION
**Pull request by alessio-acitelli**
Depends on: [INFRA PAGOPA-2421](https://github.com/pagopa/pagopa-infra/pull/2674)

#### List of Changes

- changed some configurations to prevent the advisory lock problem

#### Motivation and Context

- Error:  Failed to execute goal org.flywaydb:flyway-maven-plugin:8.0.5:migrate (default-cli) on project gpd: org.flywaydb.core.api.FlywayException: Unable to release PostgreSQL advisory lock

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: helm/values-dev.yaml [hunk 1]
```diff
@@ -86,6 +86,7 @@ microservice-chart:
     WEBSITE_SITE_NAME: 'pagopa-d-gpd-core-service'
     ENV: 'dev'
     SPRING_DATASOURCE_URL: "jdbc:postgresql://gpd-db.d.internal.postgresql.pagopa.it:6432/apd?sslmode=require&prepareThreshold=0&tcpKeepAlive=true"
+    FLYWAY_DATASOURCE_URL: "jdbc:postgresql://gpd-db.d.internal.postgresql.pagopa.it:6432/apd?sslmode=require&prepareThreshold=0&tcpKeepAlive=true&lock_timeout=30000"
```
Comment (pasqualespica) [inline helm/values-dev.yaml:89]: why add `FLYWAY_DATASOURCE_URL` if isnt'used ?`
and for flyway migration doesn't use `6432` ( pgbouncer ) but directly `5432`

File: apdFlywayConfig.conf [hunk 2]
```diff
@@ -1,4 +1,4 @@
 flyway.user=${SPRING_DATASOURCE_USERNAME}
 flyway.password=${SPRING_DATASOURCE_PASSWORD}
 flyway.schemas=${GPD_SCHEMA_NAME}
-flyway.url=${SPRING_DATASOURCE_URL}
+flyway.url=${FLYWAY_DATASOURCE_URL}
```
Comment (pasqualespica) [inline apdFlywayConfig.conf:4]: `export SPRING_DATASOURCE_URL='${{ secrets.DATASOURCE_URL }}'`
is set via `.identity`

we need to add a new one secret
```
  env_secrets = {
...
    "DATASOURCE_URL": data.azurerm_key_vault_secret.key_vault_datasource_url.value,
  }

```
something like `FLYWAY_DATASOURCE_URL`
that use dns, `5432` and new param `lock_timeout=30000` that tou hare rich out

File: helm/values-prod.yaml [hunk 3]
```diff
@@ -86,6 +86,7 @@ microservice-chart:
     WEBSITE_SITE_NAME: 'pagopa-p-gpd-core-service'
     ENV: 'prod'
     SPRING_DATASOURCE_URL: "jdbc:postgresql://gpd-db.p.internal.postgresql.pagopa.it:6432/apd?sslmode=require&prepareThreshold=0"
+    FLYWAY_DATASOURCE_URL: "jdbc:postgresql://gpd-db.p.internal.postgresql.pagopa.it:6432/apd?sslmode=require&prepareThreshold=0&lock_timeout=30000"
```
Comment (pasqualespica) [inline helm/values-prod.yaml:89]: remove it

File: helm/values-uat.yaml [hunk 4]
```diff
@@ -86,6 +86,7 @@ microservice-chart:
     WEBSITE_SITE_NAME: 'pagopa-u-gpd-core-service'
     ENV: 'uat'
     SPRING_DATASOURCE_URL: "jdbc:postgresql://gpd-db.u.internal.postgresql.pagopa.it:6432/apd?sslmode=require&prepareThreshold=0&tcpKeepAlive=true"
+    FLYWAY_DATASOURCE_URL: "jdbc:postgresql://gpd-db.u.internal.postgresql.pagopa.it:6432/apd?sslmode=require&prepareThreshold=0&tcpKeepAlive=true&lock_timeout=30000"
```
Comment (pasqualespica) [inline helm/values-uat.yaml:89]: remove it

File: pom.xml [hunk 5]
```diff
@@ -19,7 +19,7 @@
         <springdoc-version>1.6.6</springdoc-version>
         <modelmapper-version>3.1.0</modelmapper-version>
         <jackson-datatype-jsr310-version>2.13.1</jackson-datatype-jsr310-version>
-        <flyway-version>8.5.13</flyway-version>
+        <flyway-version>9.22.3</flyway-version>
```
Comment (pasqualespica) [inline pom.xml:22]: it works for pgFlex 15 vers too ?

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 5 above)
Comment (alessio-acitelli) [reply pom.xml:22]: From 9.0.5 Flyway is compatible with PostgreSQL 15:
https://documentation.red-gate.com/fd/release-notes-for-flyway-engine-179732572.html
image

Abou PGFLEX: "The compatibility includes PostgreSQL-specific features and extensions such as custom schema handling and advanced SQL functions. However, Flyway itself does not explicitly manage extensions like pgFlex. To use such extensions, you need to ensure they are installed and configured in your PostgreSQL instance prior to running Flyway migrations​"

**Review by pasqualespica — APPROVED** [review]

**Review by pasqualespica — APPROVED** [review]
lgtm 🆗

**Pull request merged**